### PR TITLE
fix: generate project code

### PR DESCRIPTION
**Pull request by dvesperini**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by dvesperini** [comment]
@kavya498 any news for the review?

**Comment by kavya498** [comment]
@dvesperini , Kindly incorporate all the PR comments provided to old PR.
Looks like you have just regenerated with few changes..

Few PR comments are not included like 
changes to config.go file.
valid examples are not provided in docs and as well as in examples.
id is not changed to project_id
testcase results are not attached.
Kindly, revisit the pr comments on old PR.

Thanks

**Comment by dvesperini** [comment]
Review comments to address:
https://github.com/IBM-Cloud/terraform-provider-ibm/pull/4447/files#r1153532801
https://github.com/IBM-Cloud/terraform-provider-ibm/pull/4447/files#r1153543083 I need Arn help
https://github.com/IBM-Cloud/terraform-provider-ibm/pull/4447/files#r1153563851
https://github.com/IBM-Cloud/terraform-provider-ibm/pull/4447/files#r1153565055

**Comment by dvesperini** [comment]
Most of changes are made.
Can anyone take a look?
By the way we won't replace id with project_id.. we had the same naming convention for all other dependecies (SDK,CLI..)
About test to run.. what is the right command to perform in local?
@kavya498

**Comment by dvesperini** [comment]
Any news??
@kavya498

File: ibm/conns/config.go [hunk 1]
```diff
@@ -1505,6 +1517,37 @@ func (c *Config) ClientSession() (interface{}, error) {
...
+		session.projectClient.Service.EnableRetries(c.RetryCount, c.RetryDelay)
+		// Add custom header for analytics
+		session.projectClient.SetDefaultHeaders(gohttp.Header{
+			"X-Original-User-Agent": {fmt.Sprintf("terraform-provider-ibm/%s", version.Version)},
+		})
+	} else {
+		session.projectClientErr = fmt.Errorf("Error occurred while configuring Projects API Specification service: %q", err)
+	}
+
+	if c.Visibility == "private" || c.Visibility == "public-and-private" {
```
Comment (kavya498) [inline ibm/conns/config.go:1547]: remove this if block..

File: examples/ibm-project/main.tf [hunk 2]
```diff
@@ -0,0 +1,36 @@
+provider "ibm" {
+  ibmcloud_api_key = var.ibmcloud_api_key
+}
+
+// Provision project_instance resource instance
+resource "ibm_project_instance" "project_instance_instance" {
```
Comment (kavya498) [inline examples/ibm-project/main.tf:6]: provide valid example

File: ibm/service/project/data_source_ibm_project_event_notification_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	`, projectName)
+}
+
+func testAccCheckIbmProjectEventNotificationDataSourceConfig(projectName string, projectDescription string, projectResourceGroup string, projectLocation string) string {
+	return fmt.Sprintf(`
+		resource "ibm_project_instance" "project_instance_instance" {
+			name = "%s"
+			description = "%s"
+			configs {
+				id = "id"
```
Comment (kavya498) [inline ibm/service/project/data_source_ibm_project_event_notification_test.go:79]: Can the value of `id` be just id?
Does API allow this? how did the testcase pass?

File: website/docs/r/project_instance.html.markdown [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+		* `event_id` - (String) The unique ID of a project.
+		  * Constraints: The maximum length is `128` characters. The value must match regular expression `/^[\\.\\-0-9a-zA-Z]+$/`.
+	* `cumulative_needs_attention_view_err` - (String) \"True\" indicates that the fetch of the needs attention items failed.
+	* `event_notifications_crn` - (String) The CRN of the event notifications instance if one is connected to this project.
+	* `location` - (String) The location where the project was created.
+	* `resource_group` - (String) The resource group where the project was created.
+	* `state` - (String) The project status value.
+	  * Constraints: The maximum length is `64` characters. The minimum length is `1` character. The value must match regular expression `/^(CREATING|CREATING_FAILED|UPDATING|UPDATING_FAILED|READY)$/`.
+
+## Provider Configuration
```
Comment (kavya498) [inline website/docs/r/project_instance.html.markdown:102]: Remove these sections of docs from this resource doc..
```
## Provider Configuration
### Static credentials
### Environment variables
```

**Review by kavya498 — CHANGES_REQUESTED** [review]

File: ibm/conns/config.go (hunk 1 above)
Comment (riccardo-angelilli) [reply ibm/conns/config.go:1547]: done

File: examples/ibm-project/main.tf (hunk 2 above)
Comment (riccardo-angelilli) [reply examples/ibm-project/main.tf]: done

File: ibm/service/project/data_source_ibm_project_event_notification_test.go (hunk 3 above)
Comment (riccardo-angelilli) [reply ibm/service/project/data_source_ibm_project_event_notification_test.go:79]: done

File: website/docs/r/project_instance.html.markdown (hunk 4 above)
Comment (riccardo-angelilli) [reply website/docs/r/project_instance.html.markdown:102]: done

**Comment by riccardo-angelilli** [comment]
output of tests:

Screenshot 2023-05-03 at 11 29 40

**Review by kavya498 — APPROVED** [review]

**Pull request merged**